lib/cisco_node_utils/syslog_server.rb

```diff
@@ -62,25 +82,64 @@ def ==(other)
     end
 
     def create
-      # Set timestamp units
-      config_set('syslog_server',
-                 'server',
-                 state: '',
-                 ip:    "#{name}",
-                 level: level.nil? ? '' : "#{level}",
-                 vrf:   vrf.nil? ? '' : "use-vrf #{vrf}",
-                )
+      if platform == :ios_xr
+
+        # This provider only support a 1-1 mapping between host and VRF.
+        # Thus, we must remove the other entries on different VRFs.
+        all_vrfs = config_get('syslog_server', 'vrf', name)
+        destroy(all_vrfs) if all_vrfs.count > 1
```

Comment:
I'm verifying manually on XR, as we don't have the minitests working yet due to a image deployment issue. I've opened a ticket for this with the relevant team.